kernel/src/snapshot.rs

```diff
@@ -422,11 +423,44 @@ mod tests {
     use object_store::path::Path;
     use object_store::ObjectStore;
 
+    use crate::actions::Format;
     use crate::engine::default::executor::tokio::TokioBackgroundExecutor;
     use crate::engine::default::filesystem::ObjectStoreFileSystemClient;
     use crate::engine::sync::SyncEngine;
     use crate::schema::StructType;
 
+    #[test]
+    fn test_snapshot_filtering() {
```

Comment:
nit: better name (is this testing P+M filter in the replay?)

Reply:
Changed to `test_replay_protocol_metadata_filtering_predicate`